pkg/contact/endpoint.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package contact
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/common/log"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+)
+
+// Endpoint implements the contact service Endpoints
+type Endpoint struct {
+	log     *zap.Logger
+	service *Service
+}
+
+// NewEndpoint returns a new contact service endpoint
+func NewEndpoint(log *zap.Logger, service *Service) *Endpoint {
+	return &Endpoint{
+		log:     log,
+		service: service,
+	}
+}
+
+// Ping provides an easy way to verify a node is online and accepting requests
+func (endpoint *Endpoint) Ping(ctx context.Context, req *pb.PingRequest) (_ *pb.PingResponse, err error) {
```

Comment:
So if this were the satellite's endpoint, inside this function there would be a client that sends a `Ping` to the SN's endpoint right?

Reply:
I clarified things somewhat when I separated the contact package into storagenode and satellite. Names of things are still up in the air- I just used placeholders for now.